### PR TITLE
test PR with my version of arkode branch with whitespace removed

### PR DESCRIPTION
**Pull request by mt5555**

### CODE REVIEW

File: components/homme/src/test_src/dcmip12_wrapper.F90 [hunk 1]
```diff
@@ -592,21 +587,14 @@ subroutine dcmip2012_test4_init(elem,hybrid,hvcoord,nets,nete)
...
         call test4_baroclinic_wave(dcmip4_moist,dcmip4_X,lon,lat,&
                                    pressure,z,zcoords,u,v,w,T,phis,ps,rho,q,q1,q2)
         qarray(1)=q; qarray(2)=q1; qarray(3)=q2;
-        !dp = pressure_thickness(ps,k,hvcoord)
-        if (k <= nlev) then
-          dp = (hvcoord%hyai(k+1)-hvcoord%hyai(k))*p0 + (hvcoord%hybi(k+1)-hvcoord%hybi(k))*ps
-        else
-          dp = (hvcoord%hyai(k)-hvcoord%hyai(k-1))*p0 + (hvcoord%hybi(k)-hvcoord%hybi(k-1))*ps
-        endif
+        dp = pressure_thickness(ps,k,hvcoord)
```
Comment (cjvogl) [inline components/homme/src/test_src/dcmip12_wrapper.F90]: When `k=nlevp`, this line will access `hyai` and `hybi` at index `nlevp+1` inside of `pressure_thickness`, which does not exist.  Perhaps a more concise change here would be to use `dp = pressure_thickness(ps,max(k,nlev),hvcoord)`

Comment (cjvogl) [inline components/homme/src/test_src/dcmip12_wrapper.F90]: When `k=nlevp`, this line will access `hyai` and `hybi` at index `nlevp+1` inside of `pressure_thickness`, which does not exist.  Perhaps a more concise change here would be to use `dp = pressure_thickness(ps,max(k,nlev),hvcoord)`

Comment (mt5555) [reply components/homme/src/test_src/dcmip12_wrapper.F90]: good catch!  sorry I broke that fix earlier.   Actually, the variable  dp should not be computed at interfaces in general, so I think the best fix would be to remove dp from set_state_i(), so we can also remove the call to pressure_thickness inside all the nlevp loops.  can you do that on your revised branch?  unfortunately there are three different versions of set_state_i in the three models that all have to be changed.

Comment (cjvogl) [reply components/homme/src/test_src/dcmip12_wrapper.F90]: No worries: this approach you suggested is much cleaner.  I went through and updated `set_state_i` in the preqx, preqx_kokkos, theta, and theta-l versions of element_ops.F90.

I'll do a force push to the other PR and we can wrap things up there (unless you want to keep using this PR).

**Pull request merged**